File: v1/src/domain/edge.ts
type EdgeType =
  | 'references'
  | 'contains'
  | 'tagged_with'
  | 'similar_to'
  | 'responds_to';

class Edge {
  readonly id: string;
  readonly sourceId: string;
  readonly targetId: string;
  readonly type?: EdgeType;
  readonly createdAt: Date;

  constructor(
    id: string,
    sourceId: string,
    targetId: string,
    type: EdgeType,
    createdAt: Date
  ) {
    this.id = id;
    this.sourceId = sourceId;
    this.targetId = targetId;
    this.type = type;
    this.createdAt = createdAt;
  }
}

export { Edge };
export type { EdgeType };

File: src/external/tui/models/edge.go
package models

import "time"

// EdgeType represents the different types of relationships between nodes
type EdgeType string

const (
	EdgeTypeReferences EdgeType = "references"
	EdgeTypeContains   EdgeType = "contains"
	EdgeTypeTaggedWith EdgeType = "tagged_with"
	EdgeTypeSimilarTo  EdgeType = "similar_to"
	EdgeTypeRespondsTo EdgeType = "responds_to"
)

// Edge represents a relationship between two nodes
type Edge struct {
	ID        string    `json:"id"`
	SourceID  string    `json:"sourceId"`
	TargetID  string    `json:"targetId"`
	Type      EdgeType  `json:"type,omitempty"`
	CreatedAt time.Time `json:"createdAt"`
}

// LinkNodesRequest represents the request to link two nodes
type LinkNodesRequest struct {
	SourceID string   `json:"sourceId"`
	TargetID string   `json:"targetId"`
	Type     EdgeType `json:"type,omitempty"`
}

// LinkNodesResponse represents the response from linking nodes
type LinkNodesResponse struct {
	OK    bool   `json:"ok"`
	Error string `json:"error,omitempty"`
}
